poi-radio/src/state.rs

```diff
@@ -85,6 +110,75 @@ impl PersistedState {
         self.remote_messages.lock().unwrap().push(msg)
     }
 
+    /// Update comparison_results
+    pub fn update_comparison_result(&self, comparison_result: ComparisonResult) {
+        let deployment = comparison_result.clone().deployment;
+
+        self.comparison_results
+            .lock()
+            .unwrap()
+            .insert(deployment, comparison_result);
+    }
+
+    /// Add entry to comparison_results
+    pub fn add_comparison_result(&self, comparison_result: ComparisonResult) {
+        let deployment = comparison_result.clone().deployment;
+
+        self.comparison_results
+            .lock()
+            .unwrap()
+            .insert(deployment, comparison_result);
+    }
+
+    pub async fn handle_comparison_result(
+        &self,
+        new_comparison_result: ComparisonResult,
+        notifier: Notifier,
+    ) {
+        let (should_notify, updated_comparison_result) = {
+            let mut results = self.comparison_results.lock().unwrap();
+            let deployment = &new_comparison_result.deployment;
+
+            if !results.contains_key(deployment) {
+                results.insert(deployment.clone(), new_comparison_result.clone());
+            }
+
+            let should_notify = match results.get(deployment) {
+                Some(current_result)
+                    if current_result.result_type != new_comparison_result.result_type
+                        && new_comparison_result.result_type != ComparisonResultType::NotFound =>
+                {
+                    // Status has changed, update and send notification
+                    results.insert(deployment.clone(), new_comparison_result.clone());
+                    true
+                }
+                Some(_) => {
+                    match new_comparison_result.result_type {
```

Comment:
maybe use an `if let` statement here to check the types since it is more concise and easier to read